insights/parsers/systemd/unitfiles.py

```diff
@@ -81,3 +93,76 @@ def is_on(self, service_name):
         """
         return self.services.get(service_name, False)
+
+
+@parser('systemctl_list-units')
+class ListUnits(Parser):
+
+    """
+    The ListUnits class parses the output of ``/bin/systemctl list-units`` and provides
+    information about all the services listed under it.
+
+    Output of Command::
+
+        sockets.target                      loaded active active    Sockets
+        swap.target                         loaded active active    Swap
+        systemd-shutdownd.socket            loaded active listening Delayed Shutdown Socket
+        neutron-dhcp-agent.service          loaded active running   OpenStack Neutron DHCP Agent
+        neutron-openvswitch-agent.service   loaded active running   OpenStack Neutron Open vSwitch Agent
+
+    Example:
+
+        >>> conf = shared[ListUnits]
+        >>> conf.get_service_details('swap.target')
+        {'LOAD': 'loaded', 'ACTIVE': 'active', 'SUB': 'active', 'UNIT': 'swap.target'}
+    """
+    def __init__(self, *args, **kwargs):
+        self.unit_list = {}
+        """dict: Dictionary service detail like active, running, exited, dead"""
+        super(ListUnits, self).__init__(*args, **kwargs)
+
+    def parse_content(self, content):
+        """
+        Main parsing class method which stores all interesting data from the content.
+
+        Args:
+            content (context.content): Parser context content
+        """
+        # 'static' means 'on' to fulfill dependency of something else that is on
+        # man systemctl - "is-enabled" knows these states
+        valid_states = {'invalid', 'loaded', 'inactive', 'active',
+                        'exited', 'running', 'failed', 'mounted', 'waiting', 'plugged'}
+
+        valid_units = {'service', 'socket', 'device', 'mount', 'automount', 'swap', 'target',
+                       'path', 'timer', 'slice', 'scope'}
+
+        for line in get_active_lines(content):
+            parts = line.split(None)  # AWK like split, strips whitespaces
+            if any(part in valid_states for part in parts):
+                service_details = {}
+                service_details['LOAD'] = parts[1]
+                service_details['ACTIVE'] = parts[2]
+                service_details['SUB'] = parts[3]
+                if any(unit in parts[0] for unit in valid_units):
+                    service_details['UNIT'] = parts[0]
+                    self.unit_list[parts[0]] = service_details
+
+    def get_service_details(self, service_name):
+        """
+        Return the service details collected by systemctl.
+
+        Args:
+            service_name (str): service name including its extension.
+
+        Returns:
+            Service details
+
+            >>> service_name = "neutron-dhcp-agent.service"
+            >>> self.data.get(service_name)
+            {'LOAD': 'loaded', 'ACTIVE': 'active', 'SUB': 'running', 'UNIT': 'neutron-dhcp-agent.service'}
+
+        Return Type:
+            Dictionary format
+        """
```

Comment:
Although this renders, a better way to format this docstring would be:
```
Returns:
    dict: Dictionary containing details for the service.  If service is not present
    dictionary values will be `None`::

        {'LOAD': loaded', 'ACTIVE': 'active', 'SUB': 'running', 'UNIT': 'neutron-dhcp-agent.service'}
```
Sphinx will automatically provide the `Return Type` section.  I don't believe an examples section is necessary since you have already shown this method in the class examples section.  Also this example uses the `data` attribute which is no longer present.